### ISSUE TITLE
I'm unable to create working custom transformer

### ISSUE THREAD

**Issue by krtr** [facebook/react-native#7083]
[react-native 0.23][windows][android]
I'm currently struggling to make typescript sourcemaps work with react-native.
As advised in https://github.com/facebook/react-native/issues/393#issuecomment-173247497,
I have created transformer: 
```js
var transformer = require('react-native/packager/transformer');
var fs = require("fs");

module.exports = function (data, callback) {
    if (data.filename.indexOf("index.android.js") !== -1) {
        callback({
            code: data.sourceCode,
            map: JSON.parse(fs.readFileSync("./index.android.js.map").toString()),
            filename: data.filename
        })
    } else {
        transformer(data, callback);
    }
};
```
and when I'm executing `react-native start --transformer <absolute path to this transformer`
after reaload js:
![error](https://cloud.githubusercontent.com/assets/8374421/14668988/ff8a0276-06e6-11e6-8028-d03d26364b9a.png)

here is repro:

* clone https://github.com/krtr/native-repro
* `npm install`
* `react-native run-android`
* `react-native start --transformer <absolute path to custom transformer`
* tap reload js in phone/emulator

in case you want recreate sourcemaps 
`tsc --target es6 --jsx react index.android.tsx`
(`npm install typescript -g` if you dont have typescript installed)

**Comment by joewood** [comment]
Hi @krtr - did you ever get this working? There seems to have been some limited success with Closure: https://github.com/mjmeintjes/boot-react-native/issues/23

**Comment by frogcjn** [comment]
Hi @krtr,
See my solution: Transformer for debugging ReactNative with TypeScript file
https://gist.github.com/frogcjn/fb33fd721e0186cc70ab47994f5c77dc

**Comment by krtr** [comment]
@joewood 

Sorry for not responding
I could not make this work, a have made workaround by compiling typescript to js (in filesystem) and watching those files with `react-native` (it's creating somewhat acceptable sourcemaps but far from optimal) 

@frogcjn 
Thank you 
In next 2 weeks I should have time to check this out

**Comment by frogcjn** [comment]
@krtr, I'll post a solution to the debug problem on react native with typescript and vscode. It is really easy and don't even need a new transformer.

**Comment by joewood** [comment]
@frogcjn can you summarize your solution? As far as I could tell the packager will not pick up local map files, so it also won't be providing them over HTTP for vscode to pick them up. 

Using a TS transform would still yield other benefits, like hot module loading support (right now the TS compiler touches all files, regardless if any changes apply).

**Comment by frogcjn** [comment]
@joewood

vscode-react-native needs some small fix for this issue:
https://github.com/Microsoft/vscode-react-native/commit/146065cd60041ad4e6416aef1ab4c806eea4ab3e

You can clone my branch and use that debugging typescript projects.

**Comment by frogcjn** [comment]
@krtr

Here is my post to explain how to debug ReactNative project in VSCode with typescript:
[make_VSCode_debug_RN_TS.md](https://gist.github.com/frogcjn/3bb47ce01a5168d8cd570a46cc71bb02)
<img width="734" alt="2016-06-07 19 31 45" src="https://cloud.githubusercontent.com/assets/1777562/15856202/bac7aeba-2ce6-11e6-9fa8-bd3d9c1a7464.png">

**Comment by krtr** [comment]
@frogcjn 
Your custom transformer works great to me :tada:

**Comment by alloy** [comment]
Here’s a custom transformer that sets up a pipeline that does TypeScript -> JS -> Babel transformation and all the source-map mangling required: https://github.com/artsy/emission/compare/93ab096089c3d07040511e070cb0585e66cccec2...5c6728b7058eac6a8fbd4353cd98e9f91a58ff99
